ir/type.cpp

```diff
@@ -1398,4 +1398,32 @@ bool isNonPtrVector(const Type &t) {
   return vty && !vty->getChild(0).isPtrType();
 }
 
+bool hasSubByte(const Type &t) {
+  if (auto agg = t.getAsAggregateType()) {
+    if (t.isVectorType()) {
+      auto &elemTy = agg->getChild(0);
+      return elemTy.isPtrType() ? false : (elemTy.bits() % 8);
+    }
+
+    for (unsigned i = 0, e = agg->numElementsConst(); i != e;  ++i) {
+      if (hasSubByte(agg->getChild(i)))
+        return true;
+    }
+  }
+  return false;
+}
+
+uint64_t getCommonAccessSize(const IR::Type &ty) {
```

Comment:
Common is not a good name, as the meaning is not immediate in this case. Perhaps `getAccessSize` is enough?

Reply:
I think `getAccessSize(ty)` can be interpreted as `sizeof(ty)`..
Or what about `getByteSize`?